src/Microsoft.Diagnostics.Monitoring/ReversedServerConnectionsSource.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Diagnostics.NETCore.Client;
+
+namespace Microsoft.Diagnostics.Monitoring
+{
+    /// <summary>
+    /// Aggregates diagnostics connections that are established at a transport path via a reversed server.
+    /// </summary>
+    internal sealed class ReversedServerConnectionsSource : IDiagnosticsConnectionsSourceInternal, IAsyncDisposable
+    {
+        private readonly Task _acceptTask;
+        private readonly CancellationTokenSource _cancellation = new CancellationTokenSource();
+        private readonly IList<ReversedDiagnosticsConnection> _connections = new List<ReversedDiagnosticsConnection>();
+        private readonly SemaphoreSlim _connectionsSemaphore = new SemaphoreSlim(1);
+        private readonly ReversedDiagnosticsServer _server;
+
+        private TaskCompletionSource<object> _newConnectionSource;
+
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Constructs a <see cref="ReversedServerConnectionsSource"/> that aggreates diagnostics connection
+        /// from a reversed diagnostics server at path specified by <paramref name="transportPath"/>.
+        /// </summary>
+        /// <param name="transportPath">
+        /// The path of the server endpoint.
+        /// On Windows, this can be a full pipe path or the name without the "\\.\pipe\" prefix.
+        /// On all other systems, this must be the full file path of the socket.
+        /// </param>
+        public ReversedServerConnectionsSource(string transportPath)
+        {
+            _server = new ReversedDiagnosticsServer(transportPath);
+
+            _acceptTask = AcceptAsync(_cancellation.Token);
```

Comment:
Consider removing things like AcceptAsync from the ctor. My concern here is that there may be some desire for control of when this happens but with things like dependency injection object creation isn't always very obvious to the service consumer.

Reply:
So you'd rather have the caller explicitly start the monitoring of server connections with another public method?

Reply:
Correct although I have not given much thought to what that api would look like,

Reply:
I'll just make it `void Start()` for now and then we can evolve it if necessary.

Reply:
I concur with Wiktor.  Perhaps a `Listen` function that starts the server listening for connections?